### DISCUSSION TITLE
Unable to resolve external symbols

### DISCUSSION THREAD

**Discussion by yearnfor** [alandefreitas/matplotplusplus#57] [Q&A]
This library is awesome. I tried to use a binary library in C++, and I added a header file and a lib file, but it prompted "Unable to resolve external symbols". Will the dynamic library and compilation tool version description be provided later?

**Comment by alandefreitas** [comment]
Hey @yearnfor ... thanks!

I don't know if this helps you but you can use the option `-DBUILD_SHARED_LIBS=ON` to compile it as a dynamic library. Something like:

```bash
cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_EXAMPLES=OFF -DBUILD_TESTS=OFF -DBUILD_SHARED_LIBS=ON
```

You should also be able to create the installer and the packages with this option on, but it's not something we really test in continuous integration. 

We usually only test it as a static library. Linking it as a static library is solving all our problems at the moment.

Sorry if I don't understand your question correctly.
